### ISSUE TITLE
one question about train.py

### ISSUE THREAD

**Issue by iszhyang** [lsaiml/CaVINet#5]
I really appreciate your work. When I read the code, I found a few puzzles.
Firstly, what do those 2 lines of code mean? (About 70 lines in train.py)
`train_labels[count, 0] = (np.where(identities == i[0]))[0][0]`
`train_labels[count, 1] = (np.where(identities == i[2]))[0][0]`
Secondly, I can't find this file `untied_softmax_weighted_error_33_33_33_lagmul_0.2_aug.h5`(about 215 lines in train.py)

**Comment by iszhyang** [comment]
What do `train_labels`'s first two fields mean?

**Comment by pvskand** [comment]
Hi. Thanks for your interest.
1) `train_labels` is in the following format : `<visual_image>, <caricature>, <label>`,  where `<label>` is 0 is visual image and caricature are of different identities and is 1 when they belong to the same person.
2) Lines 214-215 are for testing a model, you can currently uncomment it to train the model.

Hope this answers your query.

**Comment by QiuSun** [comment]
> Hi. Thanks for your interest.
> 
> 1. `train_labels` is in the following format : `<visual_image>, <caricature>, <label>`,  where `<label>` is 0 is visual image and caricature are of different identities and is 1 when they belong to the same person.
> 2. Lines 214-215 are for testing a model, you can currently uncomment it to train the model.
> 
> Hope this answers your query.

Thank you for your reply.

**Comment by iszhyang** [comment]
> Hi. Thanks for your interest.
> 
> 1. `train_labels` is in the following format : `<visual_image>, <caricature>, <label>`,  where `<label>` is 0 is visual image and caricature are of different identities and is 1 when they belong to the same person.
> 2. Lines 214-215 are for testing a model, you can currently uncomment it to train the model.
> 
> Hope this answers your query.
Thank you for your reply.
My friend and I tried to duplicate your network, but couldn't find the missing files. So we are restoring files based on your code.
Will the `../data_instances2/train/training_5.txt` file be small? Because I saw you load all the images in one go.
`train_data = np.zeros((size, 224, 224, 6), dtype=np.float32)` (about 43 lines in train.py)

**Comment by pvskand** [comment]
The data was able to fit on 12 GPU TitanX. I'll soon share the txt files for you to run. Thanks for your patience.

**Comment by iszhyang** [comment]
> The data was able to fit on 12 GPU TitanX. I'll soon share the txt files for you to run. Thanks for your patience.

I replaced `../data_instances2/train/training_5.txt` with `train-test-files/training_11.txt` and quickly ran out of RAM. Maybe it's because my computer has too little memory. I made A rough calculation that the memory requirement is 60GB when using `train-test-files/training_11.txt`.
49059(lines) * 224 * 224 * 6(image size) * 4B(numpy.float32) = 56.3GB
